### ISSUE TITLE
Get the name/email of the person who created shared album

### ISSUE THREAD

**Issue by podviaznikov** [RhetTbull/osxphotos#1122]
**Is your feature request related to a problem? Please describe.**
I have several shared albums and would like to get some basic info on the user who created them

**Describe the solution you'd like**
Return name/email of the user for each album in the report file maybe?

**Describe alternatives you've considered**
Looked through the issues here and didn't find anything.

**Additional context**

**Comment by podviaznikov** [comment]
thank you for the info on `PhotoInfo.owner`. and please check if it's possible to get email. I think it's possible in other Apple tools - like Apple Notes.

**Comment by RhetTbull** [comment]
I don't believe it's possible to get the owner email address. There's a `ZGENERICALBUM.ZCLOUDOWNEREMAILKEY` which appears to be a foreign key to another table but I cannot find a correlating table that would match this key. There's also a `ZCLOUDOWNERHASHEDPERSONID` which I'm guessing is a hashed GUID that allows Photos to lookup the person's info in iCloud.  You'll have to do with just the name.

**Comment by podviaznikov** [comment]
got it, thank you!

and this one is not related https://github.com/RhetTbull/osxphotos/issues/1130?

SharedParticipant? It's also not possible to get email from there?

**Comment by RhetTbull** [comment]
> SharedParticipant? It's also not possible to get email from there?

I don't think this is going to be possible. I'm not 100% certain as to the use of this table yet but an entry does not exist for every user who shares photos nor is email available for every participant. 

I do plan to add the name of the person who created the shared album (currently available only for the photos themselves) to the AlbumInfo class.

**Comment by podviaznikov** [comment]
@RhetTbull slightly related question. is it possible somehow to add album id or even better iCloud album id using `--keyword-template`?

I think it's already possible to get `uuid` of the photo, but I cannot find how to do that for the album.

**Comment by RhetTbull** [comment]
@podviaznikov this is not possible using the built in templates but it possible if you write a custom template function to use with the `{function}` template. See [example function](https://github.com/RhetTbull/osxphotos/blob/main/examples/template_function.py). You could access the PhotoInfo's `album_info` property which returns a list of `AlbumInfo` objects from which you could get the uuid. See the [API docs](https://github.com/RhetTbull/osxphotos/blob/main/API_README.md).  You'd then use your function something like this:

`--keyword-template {function:album_uuid.py::album_uuid}`

Keep in mind that UUIDs are unique only to a given Photo library on a given computer. The same photo or album on a different device will have a different UUID so in that sense they have limited utility. For photos in an iCloud library, there's a cloud master GUID that's accessible via `PhotoInfo.cloud_guid` or `{photo.cloud_guid}` in the template system.
